File: 2000.reverse-prefix-of-word/solution.go
func reversePrefix(word string, ch byte) string {
    i := 0
    for {
        if i == len(word) {
            return word
        }
        if word[i] == ch {
            break
        }
        i++
    }

    var r []byte
    for j := i; j >= 0; j-- {
        r = append(r, word[j])
    }    
    r = append(r, word[i+1:]...)
    return string(r)
}

File: 2000.reverse-prefix-of-word/solution.kt
class Solution {
    fun reversePrefix(word: String, ch: Char): String {
        val pos = word.indexOf(ch)
        return if (pos == -1) {
            word
        } else {
            word.subSequence(0..pos).toList().reversed().joinToString("") + word.subSequence(pos+1 until word.length) 
        }
    }
}